framework/db/QueryBuilder.php

```
@@ -1151,6 +1151,9 @@ public function buildColumns($columns)
             }
 
             $columns = preg_split('/\s*,\s*/', $columns, -1, PREG_SPLIT_NO_EMPTY);
+            if ($columns === false) {
+                return '';
```

what is the error that could occur here? I'm quite sure returning an empty string here will result in an SQL syntax error. It should throw an exception instead.

---

Yes, you are right.here sql is has syntax error if user use invalid fields or string. system will can't continue and hint use check code. Nothing or throw an exception?

---

throwing a meaningful exception is better than creating an sql error.

---

yes, Thank you. i change it .throw a InvalidParamException exception if `columns` param error.